components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/scheduler/NotificationScheduler.java

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright (c) 2011, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.humantask.core.scheduler;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.w3c.dom.Element;
+
+import org.wso2.carbon.event.output.adapter.core.OutputEventAdapterConfiguration;
+import org.wso2.carbon.event.output.adapter.core.exception.OutputEventAdapterException;
+import org.wso2.carbon.event.output.adapter.email.EmailEventAdapter;
+import org.wso2.carbon.event.output.adapter.sms.SMSEventAdapter;
+import org.wso2.carbon.humantask.TRendering;
+import org.wso2.carbon.humantask.core.HumanTaskConstants;
+import org.wso2.carbon.humantask.core.dao.TaskCreationContext;
+import org.wso2.carbon.humantask.core.dao.TaskDAO;
+import org.wso2.carbon.humantask.core.engine.runtime.api.EvaluationContext;
+import org.wso2.carbon.humantask.core.engine.runtime.api.ExpressionLanguageRuntime;
+import org.wso2.carbon.humantask.core.engine.util.CommonTaskUtil;
+import org.wso2.carbon.humantask.core.internal.HumanTaskServiceComponent;
+import org.wso2.carbon.humantask.core.store.HumanTaskBaseConfiguration;
+import org.xml.sax.InputSource;
+import org.w3c.dom.Document;
+
+import javax.xml.namespace.QName;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.StringReader;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+
+
+public class NotificationScheduler {
+    private static Log log = LogFactory.getLog(NotificationScheduler.class);
+    private boolean isNotification = false;
+    private boolean isEmailNotification = false;
+    private boolean isSMSNotification = false;
+    private boolean taskCompleted = false;
+    private static EmailEventAdapter emailAdapter;
+    private static SMSEventAdapter smsAdapter;
+    private NotificationScheduler notificationScheduler;
+    private ExecutorService exec;
+    private NotificationTask notificationTask;
+    private Map<String, String> emailDynamicProperties;
+    private Map<String, String> smsDynamicProperties;
+
+
+    private String prefix = "htd";
+    String mail_to = null;
```

Comment:
Updated in commit https://github.com/wso2/carbon-business-process/commit/c037c2607734679028dc96c59761857176b6dfdb